lib/config/configuration.go

```diff
@@ -714,7 +739,84 @@ func applyKubeConfig(fc *FileConfig, cfg *service.Config) error {
 		log.Warning("both kubernetes_service and proxy_service are enabled, but proxy_service doesn't set kube_listen_addr; consider setting kube_listen_addr on proxy_service, to handle incoming Kubernetes requests")
 	}
 	return nil
+}
+
+// applyAppsConfig applies file configuration for the "app_service" section.
+func applyAppsConfig(fc *FileConfig, cfg *service.Config) error {
+	// Apps are enabled.
+	cfg.Apps.Enabled = true
+
+	// Enable debugging application if requested.
+	cfg.Apps.DebugApp = fc.Apps.DebugApp
+
+	// Loop over all apps and load app configuration.
+	for _, application := range fc.Apps.Apps {
+		err := application.CheckAndSetDefaults()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		// Check if the application name contains spaces. Don't allow spaces because
+		// for trusted clusters the name is used to construct the vanity domain that
+		// the application will be running under.
+		if ok := strings.Contains(application.Name, " "); ok {
```

Comment:
Nit: Why not incorporate this check into `application.CheckAndSetDefaults()` above?